src/Workspaces/Remote/ServiceHub/Host/RemoteAnalyzerPathResolver.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Collections.Immutable;
+using System.Globalization;
+
+namespace Microsoft.CodeAnalysis.Remote.Diagnostics;
+
+/// <summary>
+/// For analyzers shipped in Roslyn, different set of assemblies might be used when running
+/// in-proc and OOP e.g. in-proc (VS) running on desktop clr and OOP running on ServiceHub .Net6
+/// host. We need to make sure to use the ones from the same location as the remote.
+/// </summary>
+internal sealed class RemoteAnalyzerPathResolver(string baseDirectory) : IAnalyzerPathResolver
```

Comment:
can we make this a private nsted type of whatever type ends up needing to instantiate instances of this?

Reply:
The problem is that nothing creates this except unit tests. It _seems_ like dead code but possibly this is something created via IVT or VS code? Happy to take your alls guidance here.